### PR TITLE
Add codespell support: config, github workflow (to detect future typos) + make it fix some typos

### PR DESCRIPTION
**Pull request by yarikoptic**
more on codespell: https://github.com/codespell-project/codespell

### CODE REVIEW

File: auth/browser.go [hunk 1]
```diff
@@ -63,11 +63,11 @@ func NewBrowserAuth(ctx context.Context, opts ...Option) (BrowserAuth, error) {
 	} else {
 		br.opts.workspace = wsp
 	}
-	auther, err := browser.New(br.opts.workspace, browser.OptBrowser(br.opts.browser), browser.OptTimeout(br.opts.loginTimeout), browser.OptVerbose(br.opts.verbose))
+	author, err := browser.New(br.opts.workspace, browser.OptBrowser(br.opts.browser), browser.OptTimeout(br.opts.loginTimeout), browser.OptVerbose(br.opts.verbose))
```
Comment (rusq) [inline auth/browser.go:66]: Can this be added to ignores too please?

**Review by rusq — CHANGES_REQUESTED** [review]
Hey @yarikoptic , thank you for your contribution, I quite like this, did not know that I had so many typos!  One minor comment for your consideration, please see below

File: auth/browser.go (hunk 1 above)
Comment (yarikoptic) [reply auth/browser.go:66]: oh, I should have read more of context here to understand what was wanted.
I will add it to skip.

Comment (yarikoptic) [reply auth/browser.go]: rebased, redone, force-pushed

**Review by rusq — APPROVED** [review]
LGTM, thank you @yarikoptic

**Pull request merged**